### PR TITLE
Rearange API auth tests to the correct place

### PR DESCRIPTION
**Pull request by ashb**
- tests/api/auth/bakcned/test_basic_auth was almost exclusively testing
  the Connexion API -- moved in to tests/api_connexion/ folder.

  (To make this work the conftest for connexion has been moved up a
  folder out of .../endpoints/ )
- tests/test_utils/test_remote_user_api_auth_backend.py is testing test
  code itself, and so isn't needed
- Created/copied Basic auth test to
  tests/api_connexion/test_basic_auth.py -- this ensures the Connexion
  API handles basic auth correctly too.

### CODE REVIEW

File: tests/api/auth/backend/test_basic_auth.py [hunk 1]
```diff
@@ -22,26 +22,26 @@
...
 from airflow.www.app import create_app
-from tests.test_utils.api_connexion_utils import assert_401
 from tests.test_utils.config import conf_vars
 from tests.test_utils.db import clear_db_pools
 
 
 class TestBasicAuth(unittest.TestCase):
-    def setUp(self) -> None:
+    @classmethod
+    def setUpClass(cls) -> None:
```
Comment (ashb) [inline tests/api/auth/backend/test_basic_auth.py:31]: Drive-by speed up here too.

Will do a bigger optimization PR shortly.

File: tests/api/auth/backend/test_basic_auth.py [hunk 2]
```diff
@@ -55,23 +55,10 @@ def test_success(self):
         clear_db_pools()
 
         with self.app.test_client() as test_client:
-            response = test_client.get("/api/v1/pools", headers={"Authorization": token})
+            response = test_client.get("/api/experimental/pools", headers={"Authorization": token})
             assert current_user.email == "test@fab.org"
 
         assert response.status_code == 200
-        assert response.json == {
```
Comment (ashb) [inline tests/api/auth/backend/test_basic_auth.py:62]: We're not testing the API, so we don't care about the response, just the status code.

File: tests/test_utils/test_remote_user_api_auth_backend.py [hunk 3]
```diff
@@ -1,95 +0,0 @@
...
-
-from airflow.www.app import create_app
-from tests.test_utils.config import conf_vars
-from tests.test_utils.db import clear_db_pools
-
-
-class TestGoogleOpenID(unittest.TestCase):
-    def setUp(self) -> None:
-        with conf_vars(
-            {("api", "auth_backend"): "tests.test_utils.remote_user_api_auth_backend"}
```
Comment (ashb) [inline tests/test_utils/test_remote_user_api_auth_backend.py:31]: The purpose of this test file appeared to be to test `tests.test_utils.remote_user_api_auth_backend`,  but that is "exercised" by the other Connexion tests which use this API, so we don't need to test this again.

**Review by kaxil — APPROVED** [review]
